test/MOI_additional.jl

```diff
@@ -412,3 +412,49 @@ end
     # invalid
     @test_throws ErrorException SCIP.Optimizer(some_invalid_param_name=true)
 end
+
+@testset "Query results (before/after solve)" begin
+    optimizer = SCIP.Optimizer(display_verblevel=0)
+    atol, rtol = 1e-6, 1e-6
+
+    x, y = MOI.add_variables(optimizer, 2)
+    MOI.add_constraint(optimizer, MOI.SingleVariable(x), MOI.GreaterThan(0.0))
+    MOI.add_constraint(optimizer, MOI.SingleVariable(y), MOI.GreaterThan(0.0))
+
+    c = MOI.add_constraint(
+        optimizer,
+        MOI.ScalarAffineFunction(MOI.ScalarAffineTerm.([1.0, 1.0], [x, y]), 0.0),
+        MOI.LessThan(1.0))
+
+    MOI.set(optimizer, MOI.ObjectiveFunction{MOI.ScalarAffineFunction{Float64}}(),
+            MOI.ScalarAffineFunction(MOI.ScalarAffineTerm.([1.0, 2.0], [x, y]), 0.0))
+    MOI.set(optimizer, MOI.ObjectiveSense(), MOI.MAX_SENSE)
+
+    # before optimize: can not query most results
+    @test MOI.get(optimizer, MOI.TerminationStatus()) == MOI.OPTIMIZE_NOT_CALLED
+    @test MOI.get(optimizer, MOI.PrimalStatus()) == MOI.NO_SOLUTION
+
+    @test_throws ErrorException MOI.get(optimizer, MOI.ObjectiveValue())
+    @test_throws ErrorException MOI.get(optimizer, MOI.VariablePrimal(), x)
+    @test_throws ErrorException MOI.get(optimizer, MOI.ConstraintPrimal(), c)
+    @test_throws ErrorException MOI.get(optimizer, MOI.ObjectiveBound())
+    @test_throws ErrorException MOI.get(optimizer, MOI.RelativeGap())
+    @test_throws ErrorException MOI.get(optimizer, MOI.SolveTime())
+    @test_throws ErrorException MOI.get(optimizer, MOI.SimplexIterations())
+    @test_throws ErrorException MOI.get(optimizer, MOI.NodeCount())
```

Comment:
Node count can only be queried at the end of optimization? (not usable in callbacks for instance?)

Reply:
Yeah, actually it's not necessary here, same for `SolveTime`. But it is necessary for `SimplexIterations`, it seems, so I just thought that maybe SCIP was too lax in its assertions.

But now that you mention it, it does make sense to call some of these during the solve process.
Will re-check docs.